packages/back-end/src/util/scheduleRules.ts

```diff
@@ -0,0 +1,46 @@
+import { ScheduleRule } from "../../types/feature";
+
+export const getCurrentEnabledState = (
+  scheduleRules: ScheduleRule[],
+  currentDate: Date
+) => {
+  // If a feature doesn't have any rules, don't filter it out
+  if (!scheduleRules.length) {
+    return true;
+  }
+
+  // Ensure the scheduleRules array is always sorted earliest to latest
+  const sortedRules = scheduleRules.sort((a, b) => {
+    let dateA;
+    let dateB;
+    if (a.timestamp === null) {
+      dateA = a.enableFeature ? new Date("01-01-1971") : new Date("01-01-2090"); //TODO: Find a better way to handle this
+    } else {
+      dateA = new Date(a.timestamp);
+    }
+
+    if (b.timestamp === null) {
+      dateB = b.enableFeature ? new Date("01-01-1971") : new Date("01-01-2090"); //TODO: Find a better way to handle this
+    } else {
+      dateB = new Date(b.timestamp);
+    }
+
```

Comment:
This does feel confusing. I'm having some trouble thinking of what's supposed to happen in all the different scenarios. That said, it might help to do something like this (I'm sure my logic is completely incorrect, but the idea...):
```
if (!a.timestamp && !b.timestamp) {
  return 0;
}
if (!a.timestamp) {
  return a.enableFeature ? 1 : -1;
}
if (!b.timestamp) {
 return b.enableFeature ? -1 : 1;
}
return (new Date(a.timestamp)) - (new Date(b.timestamp));
```

Reply:
I just threw this in and tested it and it worked perfectly. I'm still wrapping my head around it, and I'm going to push on this logic with some crazy use cases. Thanks for the suggestion!